miner/worker.go

```diff
@@ -1002,7 +1002,10 @@ func (w *worker) prepareWork(genParams *generateParams) (*environment, error) {
 		Coinbase:   genParams.coinbase,
 	}
 	// Set the extra field if it's allowed.
-	if !genParams.noExtra && len(w.extra) != 0 {
+	if genParams.limitExtra && len(w.extra) > 32 {
+		return nil, fmt.Errorf("invalid extra data length: %v", w.extra)
```

Comment:
This seems a bit aggressive to fail on mining. IMHO we should either refuse to start up, or if we already reached this point, we should print a warning log ang continue either truncating or leaving the field empty.

Reply:
Hmm kinda agree that we should fail earlier.
Not too aggressive though since len(extradata) <= 32 is part of the block verification rules post merge

Reply:
> Not too aggressive though

Well, if a user happened to specify 35 bytes of extradata via command line params, or API, and somehow we missed capping it earlier, it is _too_ aggressive to punish him by disabling mining, when we can just cap it at 32 and let him off with a stern warning.

Reply:
Just saw we already have the check on `minerAPI.SetExtra()`
So I think we can just remove the `limitExtra` field